### PR TITLE
tr: Accept non UTF-8 arguments for sets

### LINKED ISSUE #6343: tr: Should handle high input bytes in command-line
```console
$ echo -en '\001amp\0376\0377' | tr $'a\xfe\xffz' "01234" | hd
00000000  01 30 6d 70 31 32                                 |.0mp12|
00000006
$ echo -en '\001amp\0376\0377' | cargo run tr $'a\xfe\xffz' "01234" | hd
error: invalid UTF-8 was detected in one or more arguments

Usage: target/debug/coreutils tr [OPTION]... SET1 [SET2]

For more information, try '--help'.
```

(Original issue: https://github.com/uutils/coreutils/issues/6133)

### PR DESCRIPTION
**Pull request by RenjiSann**
Fixes #6343

### CODE REVIEW

**Comment by BenWiederhake** [comment]
That's an issue indeed.

It seems @sylvestre doesn't want to raise the MSRV, and I don't know enough on the topic to have an opinion. So let's not do that, for now.

This isn't the first time that uutils needs to convert an OsString to bytes, so I suggest taking any existing implementation (e.g. in `cut`: https://github.com/uutils/coreutils/pull/6037/files#diff-939c8f8ce2deb6faec07f4ce55505aec484179070d0c817da4f931a676c38cf4R355 ), moving it to some "canonical" place (perhaps `src/uucore/src/lib/lib.rs`?), and then just calling it. I hope this isn't too much work, and it would have the benefit of not copy-pasting the code all over the project.

So in short I'm suggesting to use `OsStrExt::as_bytes`, but without the code duplication.

**Comment by RenjiSann** [comment]
I moved `os_string_to_bytes` from `cut` to the `uucore` crate (and transformed it into `os_str_as_bytes` in the meantime), and used it inside of `tr`.

**Review by BenWiederhake — CHANGES_REQUESTED** [review]
The changes look good, the test is fine, lovely!

CI is red for a reason, please fix those issues first:
- Missing "use" on Windows: https://github.com/uutils/coreutils/actions/runs/9955233780/job/27502537737?pr=6563
- Still too high MSRV: https://github.com/uutils/coreutils/actions/runs/9955233780/job/27502538603?pr=6563
    I recommend doing `rustup default 1.70` or something like that, which should switch your toolchain down to version 1.70. Then you'll automatically notice any problems!
- Remaining CI didn't even run because of that.

**Comment by RenjiSann** [comment]
My bad, thanks for noticing

**Review by BenWiederhake — APPROVED** [review]
Yup, thank you! Remaining CI failures are known issues: #6534 #6570 (both)

I hope I'm not being too nitpicky, but can you change the first commit's subject line to follow the convention a bit closer? I suggest:
- `cut: move os_string_as_bytes utility to uucore` (mention util name)
- `tr: accept non utf8 arguments for sets` (lowercase first letter)

Thanks for fixing `tr`! :)

**Comment by RenjiSann** [comment]
> * `cut: move os_string_as_bytes utility to uucore` (mention util name)
> * `tr: accept non utf8 arguments for sets` (lowercase first letter)
> 
> Thanks for fixing `tr`! :)

Done ! :grin:
